features/dd-sdk-android-rum/src/main/kotlin/com/datadog/android/rum/internal/RumFeature.kt

```diff
@@ -251,45 +248,50 @@ internal class RumFeature(
     // region FeatureEventReceiver
 
     override fun onReceive(event: Any) {
-        if (event is JvmCrash.Rum) {
-            addJvmCrash(event)
-            return
-        } else if (event !is Map<*, *>) {
-            sdkCore.internalLogger.log(
-                InternalLogger.Level.WARN,
-                InternalLogger.Target.USER,
-                { UNSUPPORTED_EVENT_TYPE.format(Locale.US, event::class.java.canonicalName) }
-            )
-            return
-        }
-
-        when (event["type"]) {
-            NDK_CRASH_BUS_MESSAGE_TYPE ->
-                lateCrashEventHandler.handleNdkCrashEvent(event, dataWriter)
-
-            LOGGER_ERROR_BUS_MESSAGE_TYPE -> addLoggerError(event)
-            LOGGER_ERROR_WITH_STACK_TRACE_MESSAGE_TYPE -> addLoggerErrorWithStacktrace(event)
-            WEB_VIEW_INGESTED_NOTIFICATION_MESSAGE_TYPE -> {
-                (GlobalRumMonitor.get(sdkCore) as? AdvancedRumMonitor)?.sendWebViewEvent()
+        when (event) {
+            is JvmCrash.Rum -> {
+                addJvmCrash(event)
+                return
             }
 
-            TELEMETRY_ERROR_MESSAGE_TYPE -> logTelemetryError(event)
-            TELEMETRY_DEBUG_MESSAGE_TYPE -> logTelemetryDebug(event)
-            MOBILE_METRIC_MESSAGE_TYPE -> logMetric(event)
-            TELEMETRY_CONFIG_MESSAGE_TYPE -> logTelemetryConfiguration(event)
-            FLUSH_AND_STOP_MONITOR_MESSAGE_TYPE -> {
-                (GlobalRumMonitor.get(sdkCore) as? DatadogRumMonitor)?.let {
-                    it.stopKeepAliveCallback()
-                    it.drainExecutorService()
-                }
+            is InternalTelemetryEvent -> {
+                handleTelemetryEvent(event)
+                return
             }
 
-            else -> {
+            !is Map<*, *> -> {
                 sdkCore.internalLogger.log(
                     InternalLogger.Level.WARN,
                     InternalLogger.Target.USER,
-                    { UNKNOWN_EVENT_TYPE_PROPERTY_VALUE.format(Locale.US, event["type"]) }
+                    { UNSUPPORTED_EVENT_TYPE.format(Locale.US, event::class.java.canonicalName) }
                 )
+                return
+            }
+
+            else -> when (event["type"]) {
```

Comment:
I think it is better to swap these branches, and just do:

```
is Map<*, *> ->
else ->
```

reason: `!is` is harder to read and then in `else` it it takes some time to understand that type of the object there is `Map`.